piker/data/feed.py

```diff
@@ -128,30 +162,108 @@ def get_feed_bus(
 @tractor.context
 async def _setup_persistent_brokerd(
     ctx: tractor.Context,
-    brokername: str
+    brokername: str,
+
 ) -> None:
     '''
     Allocate a actor-wide service nursery in ``brokerd``
     such that feeds can be run in the background persistently by
     the broker backend as needed.
 
     '''
-    try:
-        async with trio.open_nursery() as service_nursery:
+    get_console_log(tractor.current_actor().loglevel)
 
-            # assign a nursery to the feeds bus for spawning
-            # background tasks from clients
-            bus = get_feed_bus(brokername, service_nursery)
+    global _bus
+    assert not _bus
 
-            # unblock caller
-            await ctx.started()
+    async with trio.open_nursery() as service_nursery:
+        # assign a nursery to the feeds bus for spawning
+        # background tasks from clients
+        get_feed_bus(brokername, service_nursery)
 
-            # we pin this task to keep the feeds manager active until the
-            # parent actor decides to tear it down
-            await trio.sleep_forever()
-    finally:
-        # TODO: this needs to be shielded?
-        await bus.cancel_all()
+        # unblock caller
+        await ctx.started()
+
+        # we pin this task to keep the feeds manager active until the
+        # parent actor decides to tear it down
+        await trio.sleep_forever()
+
+
+async def manage_history(
+    mod: ModuleType,
+    shm: ShmArray,
+    bus: _FeedsBus,
+    symbol: str,
+    we_opened_shm: bool,
+    some_data_ready: trio.Event,
+    feed_is_live: trio.Event,
+
+    task_status: TaskStatus = trio.TASK_STATUS_IGNORED,
+
+) -> None:
+    '''
+    Load and manage historical data including the loading of any
+    available series from `marketstore` as well as conducting real-time
+    update of both that existing db and the allocated shared memory
+    buffer.
+
+    '''
+    # TODO: history retreival, see if we can pull from an existing
+    # ``marketstored`` daemon
+    log.info('Scanning for existing `marketstored`')
```

Comment:
Woops, this should probably get commented.
Obviously these imports will come in with #247.